test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -76,6 +76,7 @@ private void assertCommandBehavior(String inputCommand,
                                       AddressBook expectedAddressBook,
                                       boolean isRelevantPersonsExpected,
                                       List<? extends ReadOnlyPerson> lastShownList) throws Exception {
+        saveFile.save(addressBook);
```

Comment:
This change doesn’t help to confirm that saving happens _only_ for mutating commands. To be complete, you should have an `if-else` branch that confirms the file is updated for mutating commands, and was not updated for non mutating commands (e.g. by checking the last updated time of the file).